pkg/apis/certmanager/v1/types_certificate.go

```diff
@@ -207,25 +207,30 @@ type CertificateSpec struct {
 
 	// Requested DNS subject alternative names.
 	// +optional
+	// +listType=set
 	DNSNames []string `json:"dnsNames,omitempty"`
```

Comment:
Some users might depend on the ordering of DNS names in their x509 certificates.

Reply:
Hmmm, does order of DNS names in a certificate matter? Maybe @SgtCoDFish knows?